#!/usr/bin/env python
# coding: utf-8

# In[40]:


# -*- coding:utf-8 -*-
# This script will download all the 10-K, 10-Q and 8-K
# provided that of company symbol and its cik code.
from __future__ import print_function  # Compatibility with Python 2

import requests
import os
import errno
from bs4 import BeautifulSoup
import datetime
import codecs
from setuptools import setup, find_packages
from setuptools.command.install import install


# In[42]:


sys.path.insert(0, os.path.abspath('.'))
SUPPORTED_VERSIONS = ['2.7', '3.5', '3.6', 'PyPy', ]
here = os.path.abspath(os.path.dirname(__file__))


# In[3]:


import sys
get_ipython().system('{sys.executable} -m pip install SECEdgar')


# In[8]:


get_ipython().system('{sys.executable} -m pip install sec-edgar-downloader')


# In[9]:


from sec_edgar_downloader import Downloader


# In[12]:


from SECEdgar.crawler import SecCrawler
import time


# In[14]:


get_ipython().system('{sys.executable} -m pip install python-edgar')


# In[25]:


import edgar
get_ipython().system('{sys.executable} -m pip install sec-edgar-download')


# In[27]:


from sec_edgar_download import indexer as ix
work_dir ='./edgar'
from_year = 2016
to_year = 2016
indexer = ix.SecIndexer(work_dir)
indexer.download_sec_feeds(from_year, to_year)


# In[29]:


cik = ix.get_cik('INTC')


# In[30]:


indexer.download_xbrl_data(cik, from_year, to_year, 'All')


# In[34]:


class EDGARQueryError(Exception):
    """
    This error is thrown when a query receives a response that is not a 200 response.
    """

    def __init__(self, response):
        self.response = response

    def __str__(self):
        return "An error occured while making the query. Received {response} response".format(
            response=self.response
        )


class EDGARFieldError(Exception):
    """
    This error is thrown when an invalid field is given to an endpoint.
    """

    def __init__(self, endpoint, field):
        self.endpoint = endpoint
        self.field = field

    def __str__(self):
        return "Field {field} not found in endpoint {endpoint}".format(
            field=self.field, endpoint=self.endpoint
        )


class CIKError(Exception):
    """
    This error is thrown when an invalid CIK is given.
    """

    def __init__(self, cik):
        self.cik = cik

    def __str__(self):
        return "CIK {cik} is not valid. Must be str or int with 10 digits.".format(cik=self.cik)


# In[39]:


DEFAULT_DATA_PATH = os.path.abspath(os.path.join(
    os.path.dirname(__file__), '..', 'SEC-Edgar-Data'))


# In[ ]:




